src/main/java/com/catapi/service/TranslationService.java

```diff
@@ -0,0 +1,124 @@
+                    });
+        });
+        log.info(STR."All active facts without translations to \{locale.getLanguageName()} are translated");
+    }
+
+    public void createCatFactTranslation(CatFact catFact, Locale locale, String translationText) {
+        CatFactTranslation catFactTranslation = new CatFactTranslation();
+        catFactTranslation.setCatFact(catFact);
+        catFactTranslation.setLocale(locale);
+        catFactTranslation.setTranslationText(translationText);
+        catFactTranslation.setUpdateMode(UpdateMode.LINGUA);
+        catFactTranslationRepository.save(catFactTranslation);
+    }
+
+    public void updateCatFactTranslation(CatFactTranslation factTranslationToUpdate, String newTranslationText) {
+        factTranslationToUpdate.setTranslationText(newTranslationText);
+        factTranslationToUpdate.setUpdateMode(UpdateMode.MANUAL);
+        catFactTranslationRepository.save(factTranslationToUpdate);
+    }
+
+    private String getLinguatoolsTranslation(Locale locale, String textToTranslate) {
+        final URI uri = UriComponentsBuilder.fromHttpUrl(LINGUATOOLS_URL)
+                .queryParam("q", textToTranslate)
+                .queryParam("google_source", "en")
+                .queryParam("google_target", locale)
+                .build().toUri();
+
+        final MediaType mediaType = MediaType.parse("text/javascript; charset=utf-8");
+        final RequestBody reqBody = RequestBody.create("Accept", mediaType);
+        final Request request = new Request.Builder()
+                .url(uri.toString())
+                .post(reqBody)
+                .addHeader("Accept", "text/javascript; charset=utf-8")
+                .build();
+
+        try (Response response = okHttpClient.newCall(request).execute()) {
+            final String responseText = Optional.ofNullable(response.body())
+                    .map(this::getStringFromRespBody)
+                    .orElseThrow(() -> new ExternalApiException("Response body cannot be null"));
+            return extractTranslation(responseText);
+        } catch (Exception e) {
+            throw new TranslationException(e.getMessage());
+        }
+    }
+
+    private String getStringFromRespBody(ResponseBody respBody) {
+        try {
+            return respBody.string();
+        } catch (IOException e) {
+            throw new ExternalApiException(e.getMessage());
+        }
+    }
+
+    private String extractTranslation(String scriptContent) {
+        final String pattern = "\\$\\('#trans_output_content'\\)\\.html\\(\"(.*?)\"\\);";
+        final Pattern r = Pattern.compile(pattern);
+        final Matcher m = r.matcher(scriptContent);
+
+        while (m.find()) {
+            String foundText = m.group(1);
+            if (!foundText.isEmpty()) {
+                return foundText;
+            }
+        }
+        throw new ExternalApiException("Wrong response format");
+    }
+}
```

Comment:
to parse linguatool repsonse. It is looking for any pattern match which is not empty in this while loop